api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/MetricHandler.java

```diff
@@ -472,14 +352,54 @@ public void findNumericData(
         ListenableFuture<NumericMetric> dataFuture = metricsService.findNumericData(metric, start, end);
         ListenableFuture<? extends Object> outputFuture = null;
         if (numberOfBuckets == 0) {
-            outputFuture = Futures.transform(dataFuture, new MetricOutMapper());
+            outputFuture = Futures.transform(dataFuture, new Function<NumericMetric, List<NumericData>>() {
+
+                @Override
+                public List<NumericData> apply(NumericMetric metric) {
+                    if (metric == null) {
+                        throw new NoResultsException();
+                    }
+
+                    return metric.getData();
+                }
+            });
         } else {
             if (bucketWidthSeconds == 0) {
                 outputFuture = Futures.transform(dataFuture, new CreateSimpleBuckets(start, end, numberOfBuckets,
                     skipEmpty));
             } else {
                 ListenableFuture<List<? extends Object>> bucketsFuture = Futures.transform(dataFuture,
-                    new CreateFixedNumberOfBuckets(numberOfBuckets, bucketWidthSeconds));
+                        new Function<NumericMetric, List<? extends Object>>() {
```

Comment:
On 10 Mar 2015, at 18:01, jsanda wrote:

>>       } else {
>>           if (bucketWidthSeconds == 0) {
>>               outputFuture = Futures.transform(dataFuture, new 
>> CreateSimpleBuckets(start, end, numberOfBuckets,
>>                   skipEmpty));
>>           } else {
>>               ListenableFuture<List<? extends Object>> bucketsFuture 
>> = Futures.transform(dataFuture,
>> -                    new CreateFixedNumberOfBuckets(numberOfBuckets, 
>> bucketWidthSeconds));
>> +                        new Function<NumericMetric, List<? extends 
>> Object>>() {
>
> The code is complicated. I checked with Mike Thompson a while back, 
> and most of it (at least at that time) was not actually used. We need 
> to find out what is actually used, get rid of the rest, and then 
> refactor. If it is not used, I don't want to keep it. It is just 
> technical debt.
>

Yes that (additional) bucketing code can go.
I implemented that some time ago to play with bucketing in the sense 
that we do e.g. not display a full 8h period of data, but only a 60 mins 
time span where all values from x hours: yy minutes in land in the 
respective yy bucket
to better see if there are repeating patterns like a spike on minute 5 
each hour.

-- 
hwr@pilhuhn.de  m:0179/207 4919  b:http://pilhuhn.blogspot.com